6211 Тягун АР Лаб.1,2

HLLDRGN opened the pull request:
Прикрепил первую  и вторую лабораторную работу

HLLDRGN committed:
Create .gitkeep

HLLDRGN committed:
6211 ТягунАР Лаб.1

Первая сдача лабораторной работы

HLLDRGN committed:
Create .gitkeep

HLLDRGN committed:
6211 Тягун АР Лаб.2/Вар.24

HLLDRGN committed:
Delete Лабораторная работа 2 directory

HLLDRGN committed:
Create .gitkeep

HLLDRGN committed:
6211 ТягунАР Лаб.2/Вар.24

mxwrlld reviewed (commented):
Общие моменты: 
- Форматируйте. Рекомендую настроить форматтер. Например, Prettier.
- Переименуйте классы согласно БЭМ: `имя-блока__имя-элемента_имя-модификатора`. Подробнее [тут](https://habr.com/ru/companies/yandex/articles/276035/) и [тут](https://ru.bem.info/methodology/quick-start/#%D1%8D%D0%BB%D0%B5%D0%BC%D0%B5%D0%BD%D1%82). Например, `cell -> maze__cell` и т.д.

Первая лабораторная:
- Сделайте лабиринт более похожим на лабиринт, сейчас он напоминает решетку. При отрисовке лабиринта используйте классы - модификаторы элемента maze__cell, которые будут скрывать / отрисовывать границы ячейки. Например, класс, отвечающий за отрисовку верхней границы можно назвать `maze__cell_border-top`. Если требуется отрисовать несколько границ, комбинируйте классы - модификаторы, например, `<div class="maze__cell maze__cell_border-top maze__cell_border-left">1</div>`. Т.е. `maze__cell` содержит общие свойства ячеек, а специфические свойства в виде отрисованных / скрытых границ задаются отдельными классами. Подход с n-м ребенком плохо масштабируется: при 100 ячейках CSS - файл значительно расширится. Сейчас лабиринт выглядит так, открывался в Chrome: <img width="1285" height="716" alt="Image" src="https://github.com/user-attachments/assets/2787d460-5064-46dd-86cd-dc4e12fc3188" />

Вторая лабораторная:
- Во втором задании подход с child-1, child-2 и т.д.. также плохо масштабируется. Плюс, затруднительно воспринимается связь между детьми и родителями. Попробуйте иной подход, согласуемый с БЭМ:
    - Есть общий класс блока родителя `parent`, конкретный родитель задается через модификатор `parent_1`. 
    - Дети - элементы блока родителя также имеют общий класс `parent__child`. Учитывая, что дети одного родителя имеют общие стили, их логично выделить в общий класс, логично назвать его с намеком на принадлежность ребенка к конкретному родителю, например, `parent__child_origin_parent-1`. При таком подходе прописывать детей - гостей становится проще
```
<div class="parent parent_1">
    Родитель 1
    <div class="parent__child parent__child_origin_parent-1">Ребенок 1</div>
    <div class="parent__child parent__child_origin_parent-1">Ребенок 2</div>
    <div class="parent__child parent__child_origin_parent-1">Ребенок 3</div>
</div>
```
- Также во втором задании нарушается условие "И третий “родитель” и его “дети” цвета green, но только если они идут после вторых, если же они идут между первыми и вторыми, то они стандартного цвета;". Если дети третьего родителя идут до второго должны быть черными. <img width="2444" height="569" alt="Image" src="https://github.com/user-attachments/assets/ccf08cf3-76c6-4536-98ba-29e3414d5923" />

mxwrlld reviewed (changes requested):
Напоминаю о необходимости внесения изменений

HLLDRGN committed:
Изменение лабиринта

Изменил код и  css, чтобы лабиринт стал более похожим на лабиринт

HLLDRGN committed:
Исправление недочетов 1 ЛР

Форматировал название классов согласно БЭМ, а также сделал лабиринт более похожим на лабиринт

HLLDRGN committed:
Update laba2.html

Внес изменения согласно общим моментам с форматированием, а также отдельные корректировки с заданным про детей и родителей

HLLDRGN committed:
Update styles.css

Внесение изменений в CSS файл

HLLDRGN reviewed (commented):
Исправил первую и вторую лабораторную работы

mxwrlld reviewed (changes requested):
Первая зачтена! 
:exclamation: Во второй работе в CSS - файле находится HTML, стилей нет :exclamation:

HLLDRGN commented:
Скорее всего, когда заливал ошибся, сейчас исправлю

HLLDRGN committed:
Update styles.css

Исправил)

mxwrlld commented:
Вторая зачтена!
